src/ucs/sys/sock.c

```diff
@@ -890,3 +878,26 @@ ucs_status_t ucs_sockaddr_get_ip_local_port_range(ucs_range_spec_t *port_range)
 
     return UCS_OK;
 }
+
+ucs_status_t ucs_sockaddr_get_ipstr(const struct sockaddr *sock_addr, char *str,
+                                    size_t max_size)
+{
+    if (sock_addr == NULL) {
+        ucs_strncpy_zero(str, "<null>", max_size);
+        return UCS_ERR_INVALID_PARAM;
```

Comment:
IMO , it's weird that the function returns error but still initializes "str" to a string..

Reply:
we did it here to handle RDMACM case when it passes `NULL`
https://github.com/openucx/ucx/pull/6488